helper-cli/src/main/kotlin/commands/CreateAnalyzerResultFromPackageListCommand.kt

```diff
@@ -0,0 +1,168 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.helper.commands
+
+import com.fasterxml.jackson.module.kotlin.readValue
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.file
+
+import org.ossreviewtoolkit.helper.utils.writeOrtResult
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Repository
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.Excludes
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.config.ScopeExclude
+import org.ossreviewtoolkit.model.config.ScopeExcludeReason
+import org.ossreviewtoolkit.model.mapper
+import org.ossreviewtoolkit.model.orEmpty
+import org.ossreviewtoolkit.utils.common.expandTilde
+import org.ossreviewtoolkit.utils.ort.Environment
+
+internal class CreateAnalyzerResultFromPackageListCommand : CliktCommand(
+    "A command which turns a simple definition file into an analyzer result."
+) {
+    private val packageListFile by option(
+        "--package-list-file", "-i",
+        help = "The package list file to read the packages metadata and the project metadata from."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val ortFile by option(
+        "--ort-file", "-o",
+        help = "The ORT file to write the generated analyzer result to."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        val packageList = packageListFile.mapper().readValue<PackageList>(packageListFile)
```

Comment:
Our goal is still to migrate to migrate to kotlinx.serialization (#3904) so please do not introduce new code that depends on Jackson. Also, we want to migrate to use camel-case in all configuration files (#5715) so please do not introduce a new file format that uses snake-case.

Reply:
> please do not introduce new code that depends on Jackson.

I was thinking the same, but then if we reuse existing model classes that have Jackson annotations, we might depend in Jackson here.

Reply:
Ok, I just read [your comment below](https://github.com/oss-review-toolkit/ort/pull/7305#discussion_r1319797722). Indeed, if this code would have been using kotlinx-serialization to begin with we could have avoided the discussion about reusing model classes.

Let's go with dedicated classes then, but still with nesting, and their layout should resemble the existing model classes as much as possible.

Reply:
I've adjusted the code to use camel case instead of snake case. 
The code is still using jackson as I prefer not to put in the effort to migrate to kotlinx serialization right now, just because there is an open ticket. I believe it's a more appropriate approach to do so in scope of that migration ticket instead of blocking any incoming PRs which use an approach consistent with the de facto standard approach in ORT.

Reply:
> which use an approach consistent with the de facto standard approach in ORT

Most code that deserializes external files has been migrated to kotlinx.serialization already (clients, scanners, some package managers). Btw, in this case migration would be trivial.